frontend/src/components/payment/transactions/TransactionList.tsx

```diff
@@ -0,0 +1,99 @@
+import { useTranslation } from 'next-i18next'
+import Link from 'next/link'
+import { FunctionComponent, useState } from 'react'
+import { Transaction } from '../../../types/Payment'
+import Button from '../../Button'
+import TransactionCancelButton from './TransactionCancelButton'
+import styles from './TransactionList.module.scss'
+
+interface ListProps {
+  transactions: Transaction[]
+}
+
+const TransactionList: FunctionComponent<ListProps> = ({ transactions }) => {
+  const { t } = useTranslation()
+
+  return (
+    <table className={styles.transactionList}>
+      <thead>
+        <tr>
+          <td>{t('created')}</td>
+          <td>{t('updated')}</td>
+          <td>{t('id')}</td>
+          <td>{t('type')}</td>
+          <td>{t('value')}</td>
+          <td>{t('status')}</td>
+          <td>{t('actions')}</td>
+        </tr>
+      </thead>
+      <tbody>
+        {transactions.length ? (
+          transactions.map((transaction) => (
+            <TransactionInfo key={transaction.id} transaction={transaction} />
+          ))
+        ) : (
+          <tr>
+            <td>{t('no-transaction-history')}</td>
+          </tr>
+        )}
+      </tbody>
+    </table>
+  )
+}
+
+interface RowProps {
+  transaction: Transaction
+}
+
+const TransactionInfo: FunctionComponent<RowProps> = ({ transaction }) => {
```

Comment:
I would prefer this to be it's own file, just cause I'm used to that and dislike multiple components sharing a file. Even if it's only used from this one.